test/ecto/repo/autogenerate_test.exs

```diff
@@ -220,6 +242,93 @@ defmodule Ecto.Repo.AutogenerateTest do
     assert default.updated_at == naive_datetime
   end
 
+  test "updates updated_at when there are children records with changes" do
+    company =
+      TestRepo.insert!(%Company{
+        updated_at: ~N[2000-01-01 00:00:00],
+        offices: [%Office{id: 1, name: "1"}]
+      })
+
+    changes = %{offices: [%{id: 1, name: "updated"}]}
+
+    updated_company =
+      company
+      |> Ecto.Changeset.cast(changes, [])
+      |> Ecto.Changeset.cast_assoc(:offices)
+      |> TestRepo.update!()
+
+    assert company.updated_at != updated_company.updated_at
+  end
+
+  test "updates updated_at when there are parent records with changes" do
```

Comment:
this test is passing, but because of the wrong reasons and I didn't figure out why.
In line https://github.com/elixir-ecto/ecto/pull/3903/files#diff-91ed3e8c550063fcf5e2df36020d92405a303c5f5745c0bf643c7dc4c538befbR447, there are always changes in the `original` variable: `%{company_id: 1}`
I'm not sure why there is always that value there, but it means that is not going through the path of the new code